### DISCUSSION TITLE
Sir: Does harfbuzz giving an API to get correct supporting tables and features inside a font file? Or Are there exist any example of using opentype features?

### DISCUSSION THREAD

**Discussion by iamjustadd** [harfbuzz/harfbuzz#2950] [Q&A] [answered]
Excuse me sir, Im using harfbuzz to development a shape engine. And when I want to use some font features ,like kerning , by
harfbuzz. I dont really found a use case of opentype feautres in harfbuzz development document, and I can't found a
use case in google either.
I read Skia code, which use harfbuzz in SkShaper_harfbuzz.cpp but it seems they also don't support opentype features.
Does harfbuzz offers API that can read tables and return how much feature tables in a (.otf or .ttf) file.

**Comment by behdad** [comment, marked as answer]
HarfBuzz automatically enables default features. And the `hb_shape()` API gives you a way to enable / disable features. What are you trying to do?

**Comment by iamjustadd** [comment]
Fine ,Maybe  i got be missleaded. 
Because I found that figma offer an interesting feature that user can set to enable or disable the font features by using the feature lists they given. So I was though that maybe there are ways to extract features list from a font files, so that I can get and print such list.
My bad, lol.

**Reply by khaledhosny** [reply]
See https://harfbuzz.github.io/harfbuzz-hb-ot-layout.html

**Comment by iamjustadd** [comment]
I have no more questions ,thank you sir.  
ME ME DA:)
